src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -152,58 +172,278 @@ export default abstract class ComparisonStore extends AnalysisStore
+                        ) || this.showSurvivalTab
+                    );
+                    this.tabHasBeenShown.set(
+                        GroupComparisonTab.MRNA,
+                        !!this.tabHasBeenShown.get(GroupComparisonTab.MRNA) ||
+                            this.showMRNATab
+                    );
+                    this.tabHasBeenShown.set(
+                        GroupComparisonTab.PROTEIN,
+                        !!this.tabHasBeenShown.get(
+                            GroupComparisonTab.PROTEIN
+                        ) || this.showProteinTab
+                    );
+                    this.tabHasBeenShown.set(
+                        GroupComparisonTab.DNAMETHYLATION,
+                        !!this.tabHasBeenShown.get(
+                            GroupComparisonTab.DNAMETHYLATION
+                        ) || this.showMethylationTab
+                    );
+                    this.tabHasBeenShown.set(
+                        GroupComparisonTab.GENERIC_ASSAY_PREFIX,
+                        !!this.tabHasBeenShown.get(
+                            GroupComparisonTab.GENERIC_ASSAY_PREFIX
+                        ) || this.showGenericAssayTab
+                    );
+                    this.tabHasBeenShown.set(
+                        GroupComparisonTab.GENERIC_ASSAY_BINARY_PREFIX,
+                        !!this.tabHasBeenShown.get(
+                            GroupComparisonTab.GENERIC_ASSAY_BINARY_PREFIX
+                        ) || this.showGenericAssayBinaryTab
+                    );
+                    this.tabHasBeenShown.set(
+                        GroupComparisonTab.ALTERATIONS,
+                        !!this.tabHasBeenShown.get(
+                            GroupComparisonTab.ALTERATIONS
+                        ) || this.showAlterationsTab
+                    );
+                    this.tabHasBeenShown.set(
+                        GroupComparisonTab.MUTATIONS,
+                        !!this.tabHasBeenShown.get(
+                            GroupComparisonTab.MUTATIONS
+                        ) || this.showMutationsTab
+                    );
+                })
+            );
+
+            this.reactionDisposers.push(
+                reaction(
+                    () => this.isSessionLoaded,
+                    isComplete => {
+                        if (isComplete) {
+                            this.loadCustomSurvivalCharts();
+                        }
+                    }
+                )
+            );
         }); // do this after timeout so that all subclasses have time to construct
     }
 
-    public destroy() {
-        this.tabHasBeenShownReactionDisposer &&
-            this.tabHasBeenShownReactionDisposer();
+    public destroy(): void {
+        for (const disposer of this.reactionDisposers) {
+            disposer();
+        }
+    }
+
+    @observable
+    public startEventPosition: 'FIRST' | 'LAST' = 'FIRST';
```

Comment:
i think this needs to be someone scoped better.  i.e. it needs to be somehow associated with the survival plot stuff, not just hanging out here in top level of store.